### DISCUSSION TITLE
Crazy Cataclysm widget modification

### DISCUSSION THREAD

**Discussion by NetSysFire** [CleverRaven/Cataclysm-DDA#57956] [Ideas]
First of all, I hope to summon @wapcaplet, the jsonifier of the sidebar widgets, here since I think they can help me. This is most definitely not worthy of an issue, so I am using a discussion instead.

My plan is as follows:

Crazy Cataclysm replaces "very obese" with "Husky".

https://github.com/CleverRaven/Cataclysm-DDA/blob/424215fc731eb29fd1f02c730a413996b4b1a631/data/mods/CrazyCataclysm/crazy_widgets.json#L73-L80

I want to replace that with "Persian" when the character has crossed the feline mutation threshold to make Crazy Cataclysm even more crazy. And maybe also replace "Husky" with "Good boi" when the player has crossed the lupine mutation threshold. But as far as I can see there is no variable for this and since this is more or less just a cosmetic minor edge case (like the mood faces), I do not even know if it is worth exposing another variable for something silly like this.

**Comment by dseguin** [comment]
You can do this with conditional clauses. That's actually how these were JSON-ified in the first place, and it's really easy to expand.

You could do that with something like this:

```diff
diff --git a/data/mods/CrazyCataclysm/crazy_widgets.json b/data/mods/CrazyCataclysm/crazy_widgets.json
index ea2484837a..3ac45dd2bd 100644
--- a/data/mods/CrazyCataclysm/crazy_widgets.json
+++ b/data/mods/CrazyCataclysm/crazy_widgets.json
@@ -76,7 +76,33 @@
         "condition": {
           "and": [
             { "compare_int": [ { "u_val": "bmi_permil" }, ">", { "const": 35000 } ] },
-            { "compare_int": [ { "u_val": "bmi_permil" }, "<=", { "const": 40000 } ] }
+            { "compare_int": [ { "u_val": "bmi_permil" }, "<=", { "const": 40000 } ] },
+            { "not": { "u_has_trait": "THRESH_FELINE" } },
+            { "not": { "u_has_trait": "THRESH_LUPINE" } }
+          ]
+        }
+      },
+      {
+        "id": "very_obese",
+        "text": "Persian",
+        "color": "red",
+        "condition": {
+          "and": [
+            { "compare_int": [ { "u_val": "bmi_permil" }, ">", { "const": 35000 } ] },
+            { "compare_int": [ { "u_val": "bmi_permil" }, "<=", { "const": 40000 } ] },
+            { "u_has_trait": "THRESH_FELINE" }
+          ]
+        }
+      },
+      {
+        "id": "very_obese",
+        "text": "Good boi",
+        "color": "red",
+        "condition": {
+          "and": [
+            { "compare_int": [ { "u_val": "bmi_permil" }, ">", { "const": 35000 } ] },
+            { "compare_int": [ { "u_val": "bmi_permil" }, "<=", { "const": 40000 } ] },
+            { "u_has_trait": "THRESH_LUPINE" }
           ]
         }
       },
```

See [the documentation](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/NPCs.md#dialogue-conditions) for a (almost) full list of possible conditions.

**Reply by NetSysFire** [reply]
Thanks! I looked in the wrong place it seems. I looked into WIDGETS.md and also condition.cpp but seems like I was on the entirely wrong track.
